### PR TITLE
allow shift+<key> when naming savegame

### PR DESCRIPTION
**Pull request by Phytolizer**
useful for symbols such as ()

### CODE REVIEW

File: prboom2/src/m_menu.c [hunk 1]
```diff
@@ -5000,7 +5000,10 @@ dboolean M_Responder (event_t* ev) {
     }
     else if (ch > 0)
     {
-      ch = toupper(ch);
+      if (shiftdown && ch >= 32 && ch <= 127)
```
Comment (JadingTsunami) [inline prboom2/src/m_menu.c]: I suppose you could move this inside the `if` below since the range check is done by that point.

**Comment by Phytolizer** [comment]
Future work (not this PR): add full text entry akin to the console

This was mentioned on Discord, replicating it here for accessibility.

**Comment by kraflab** [comment]
I'll probably take a look at full text entry prior to the next release (actually, I'm gonna be taking a look at the whole menu system and input is just a subset of that) and will hold onto this as a fallback if it ends up being too problematic.

**Comment by kraflab** [comment]
Well here's one problem, the font drawing these strings is incomplete, so only some symbols work 😅

**Comment by Phytolizer** [comment]
Hmm... is there a "default" glyph like the tofu in modern fonts? Maybe it could draw that as a failsafe

**Comment by kraflab** [comment]
By default it draws an `!`. I tested out falling back on the hud font and it's pretty reasonable I think. It just affects a few symbols. There's a bigger problem though, which is that none of this works for heretic / hexen. There are 2 issues: their text drawing code works differently, assuming contiguous font data in the wad (so drawing some symbols causes an overrun into junk data), and since they never expected people to type symbols, the font has symbols in the wrong spots (I think `[` is `_` for instance).

**Comment by kraflab** [comment]
Oh actually, if you type to enter shift + ] on master it actually crashes in heretic, so this is already an issue 😂

**Comment by kraflab** [comment]
https://github.com/kraflab/dsda-doom/commit/c96b49addc8866305acefc71c42e3e3f3c283337 and https://github.com/kraflab/dsda-doom/commit/22e6580120602adbefab4e22fa6df5d0ad891348 fyi.

Going to need to set up guard clauses for raven or set it up to use the existing font arrays instead of diving into the lumps haphazardly.

**Comment by kraflab** [comment]
PR committed here, excluding raven until a full solution is in place for that: https://github.com/kraflab/dsda-doom/commit/6334d11ff6dcb87b6fcfbf36dbea8ac98f7cd29f

Thanks!

**Pull request closed without merging**